### PR TITLE
fix: nyc command with shard option

### PR DESCRIPTION
**Pull request by bryanjtc**
Fixes: #256 
Added a new environment variable to check if the shard cli option is used to run the nyc command conditionally.

<details>
  <summary>📦 Published PR as canary version: <code>0.10.1--canary.257.aa7a0c9.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/test-runner@0.10.1--canary.257.aa7a0c9.0
  # or 
  yarn add @storybook/test-runner@0.10.1--canary.257.aa7a0c9.0
  ```
</details>

### CODE REVIEW

**Comment by yannbf** [comment]
Hey @bryanjtc thank you so much for this contribution (and the many others)!

Does this mean that for developers using `shard` option, they should figure out how to do the reporting themselves?

**Comment by bryanjtc** [comment]
> Hey @bryanjtc thank you so much for this contribution (and the many others)!
> 
> Does this mean that for developers using `shard` option, they should figure out how to do the reporting themselves?

Yes. Especifically the coverage json file for each ci job should be renamed using the shardIndex value.
Then a new ci job has to merge this coverage.json files using the nyc merge command to output the complete report.

**Comment by yannbf** [comment]
> 

Could you assist in writing some sort of recipe? Also there should be extra changes in `reportCoverage` I assume? To have the renaming bit right

**Comment by bryanjtc** [comment]
@yannbf Done. The renaming part should be done in the yml ci file. I added a recipe for github, gitlab and circle ci.

File: README.md [hunk 1]
```diff
@@ -394,6 +395,62 @@ Here's an example on how to achieve that:
...
+
+```yml
+test:
+  name: Running Test-storybook (${{ matrix.shard }})
+  strategy:
+    matrix:
+      shard: [1, 2, 3, 4]
+  steps:
+    - name: Testing storybook
+      uses: yarn test-storybook --coverage --shard=${{ matrix.shard }}/${{ strategy.job-total }}
```
Comment (MH4GF) [inline README.md:412]: This place should probably be run

```suggestion
      run: yarn test-storybook --coverage --shard=${{ matrix.shard }}/${{ strategy.job-total }}
```

**Comment by yannbf** [comment]
You're the best @bryanjtc, thanks a lot for this contribution! There were some small issues regarding watch mode so I added a few fixes as well.

**Pull request merged**